### PR TITLE
Add incremental backup test for groups

### PR DESCRIPTION
**Pull request by meain**
---

#### Does this PR need a docs update or release note?

- [ ] :white_check_mark: Yes, it's included
- [ ] :clock1: Yes, but in a later PR
- [x] :no_entry: No

#### Type of change

- [ ] :sunflower: Feature
- [ ] :bug: Bugfix
- [ ] :world_map: Documentation
- [x] :robot: Supportability/Tests
- [ ] :computer: CI/Deployment
- [ ] :broom: Tech Debt/Cleanup

#### Issue(s)

* closes https://github.com/alcionai/corso/issues/3990

#### Test Plan

- [ ] :muscle: Manual
- [ ] :zap: Unit test
- [x] :green_heart: E2E

### CODE REVIEW

File: src/internal/m365/collection/exchange/metadata.go [hunk 1]
```diff
@@ -15,13 +15,13 @@ import (
 
 // MetadataFileNames produces the category-specific set of filenames used to
 // store graph metadata such as delta tokens and folderID->path references.
-func MetadataFileNames(cat path.CategoryType) []string {
+func MetadataFileNames(cat path.CategoryType) [][]string {
```
Comment (ryanfkeepers) [inline src/internal/m365/collection/exchange/metadata.go:18]: Okay, so, I think I grok _why_ this change is wanted.  But I'm not seeing where the changes are consumed.  I'd expect some caller of this func would need to change its behavior based on the return value, or that the func it passes that return value into would need to change, but I'm not seeing either of those.  Am I missing it?  Where is this double slice utilized?

Comment (meain) [reply src/internal/m365/collection/exchange/metadata.go:18]: This change is consumed by [`checkMetadataFileExists`](https://github.com/alcionai/corso/blob/0748e138bc6c234b8ee9462875369721af486f96/src/internal/operations/test/helper_test.go#L273). The signature for that has changed in this PR. This change is also reflected [here](https://github.com/alcionai/corso/blob/0748e138bc6c234b8ee9462875369721af486f96/src/internal/operations/test/exchange_test.go#L246).

Comment (ryanfkeepers) [reply src/internal/m365/collection/exchange/metadata.go:18]: This is _only_ used in testing?  That feels... weird.  Going to approve to unblock, but I think that func should be moved into the test package if its not used in live code.

**Review by ryanfkeepers — APPROVED** [review]

File: src/internal/m365/collection/exchange/metadata.go (hunk 1 above)
Comment (meain) [reply src/internal/m365/collection/exchange/metadata.go:18]: Let me take it up separately.

**Pull request merged**